### ISSUE TITLE
Crash trying to add a new point

### ISSUE THREAD

**Issue by ShelleyLake** [BourgeoisLab/GPXLab#20]
@BourgeoisLab I've built the latest release available here (0.7.0) with Qt 5.12.2.
I experience a crash _every_ time i try to add a new point to the attached GPX.
Can you reproduce the problem?

[2019-10-27T10_53_03Z.zip](https://github.com/BourgeoisLab/GPXLab/files/3835286/2019-10-27T10_53_03Z.zip)

**Comment by sikmir** [comment]
Could you please share core dump printout? I can't reproduce crash on my Linux machine.

**Comment by ShelleyLake** [comment]
After more testing I realised that the crash happens only when working on the active file shown when starting GPXLAB. Opening/reopening the file does not crash.

**Test 1**
1. Start GPXLAB
2. Open the file
3. Insert point
4. **OK** 
5. Close

**Test 2**
1. Start GPXLAB
2. The previous file is _already shown_, so I **do not** open it again
3. Insert point
4. **Crash**

```
 	Qt5Guid.dll!00007ffef8416e6e()	Unknown
 	Qt5Guid.dll!00007ffef86fbc54()	Unknown
 	Qt5Guid.dll!00007ffef82918f1()	Unknown
>	GPXLab.exe!QMapTrack::drawPoint(QPainter * painter, const qmapcontrol::MapAdapter * mapadapter, const QRect & viewport, const QPixmapExt * pixmap, double latitude, double longitude) Line 294	C++
 	GPXLab.exe!QMapTrack::draw(QPainter * painter, const qmapcontrol::MapAdapter * mapadapter, const QRect & screensize, const QPoint offset) Line 380	C++
 	qmapcontrol0.dll!qmapcontrol::Layer::drawYourGeometries(QPainter * painter, const QPoint mapmiddle_px, QRect viewport) Line 290	C++
 	qmapcontrol0.dll!qmapcontrol::Layer::drawYourImage(QPainter * painter, const QPoint mapmiddle_px) Line 271	C++
 	qmapcontrol0.dll!qmapcontrol::LayerManager::newOffscreenImage(bool clearImage, bool showZoomImage) Line 316	C++
 	qmapcontrol0.dll!qmapcontrol::LayerManager::forceRedraw() Line 530	C++
 	qmapcontrol0.dll!qmapcontrol::MapControl::updateRequestNew() Line 432	C++
 	GPXLab.exe!QMapWidget::selectTrack(int trackNumber) Line 312	C++
 	GPXLab.exe!GPXLab::trackSelectionChanged(int trackNumber, int trackSegmentNumber, int pointNumber, const GPX_wptType * trkpt) Line 661	C++
 	GPXLab.exe!GPXLab::qt_static_metacall(QObject * _o, QMetaObject::Call _c, int _id, void * * _a) Line 350	C++
 	Qt5Cored.dll!00007ffef79656f6()	Unknown
 	Qt5Cored.dll!00007ffef7964e38()	Unknown
 	GPXLab.exe!GPX_wrapper::trackSelectionChanged(int _t1, int _t2, int _t3, const GPX_wptType * _t4) Line 426	C++
 	GPXLab.exe!GPX_wrapper::select(int trackNumber, int trackSegmentNumber, int pointNumber, bool forceSelect) Line 1056	C++
 	GPXLab.exe!GPX_wrapper::insertPoint(int trackNumber, int & trackSegmentNumber, int pointNumber, const GPX_wptType & wpt) Line 999	C++
 	GPXLab.exe!PointInsertCommand::redo() Line 40	C++
 	Qt5Widgetsd.dll!00007ffefb90100d()	Unknown
 	GPXLab.exe!GPXLab::on_actionInsert_Point_triggered() Line 251	C++
 	GPXLab.exe!GPXLab::qt_static_metacall(QObject * _o, QMetaObject::Call _c, int _id, void * * _a) Line 398	C++
 	GPXLab.exe!GPXLab::qt_metacall(QMetaObject::Call _c, int _id, void * * _a) Line 437	C++
 	Qt5Cored.dll!00007ffef791ce24()	Unknown
 	Qt5Cored.dll!00007ffef79657e0()	Unknown
 	Qt5Cored.dll!00007ffef7964e38()	Unknown
 	Qt5Widgetsd.dll!00007ffefb25e7cd()	Unknown
 	Qt5Widgetsd.dll!00007ffefb25db32()	Unknown
 	Qt5Widgetsd.dll!00007ffefb557283()	Unknown
 	Qt5Widgetsd.dll!00007ffefb557170()	Unknown
 	Qt5Widgetsd.dll!00007ffefb54e870()	Unknown
 	Qt5Widgetsd.dll!00007ffefb2eb6cd()	Unknown
 	Qt5Widgetsd.dll!00007ffefb550db9()	Unknown
 	Qt5Widgetsd.dll!00007ffefb28c15e()	Unknown
 	Qt5Widgetsd.dll!00007ffefb287408()	Unknown
 	Qt5Cored.dll!00007ffef790e236()	Unknown
 	Qt5Cored.dll!00007ffef790e0db()	Unknown
 	Qt5Widgetsd.dll!00007ffefb28e8b4()	Unknown
 	Qt5Widgetsd.dll!00007ffefb3363f0()	Unknown
 	Qt5Widgetsd.dll!00007ffefb335466()	Unknown
 	Qt5Widgetsd.dll!00007ffefb28c15e()	Unknown
 	Qt5Widgetsd.dll!00007ffefb286bff()	Unknown
 	Qt5Cored.dll!00007ffef790e236()	Unknown
 	Qt5Cored.dll!00007ffef790e0db()	Unknown
 	Qt5Guid.dll!00007ffef8311194()	Unknown
 	Qt5Guid.dll!00007ffef8313f13()	Unknown
 	Qt5Guid.dll!00007ffef82e2335()	Unknown
 	qwindowsd.dll!00007fff081205d2()	Unknown
 	Qt5Cored.dll!00007ffef79cf008()	Unknown
 	user32.dll!00007fff59c863ed()	Unknown
 	user32.dll!00007fff59c85de2()	Unknown
 	Qt5Cored.dll!00007ffef79cf8f8()	Unknown
 	qwindowsd.dll!00007fff08120584()	Unknown
 	Qt5Cored.dll!00007ffef7908463()	Unknown
 	Qt5Cored.dll!00007ffef790869e()	Unknown
 	Qt5Widgetsd.dll!00007ffefb54c06f()	Unknown
 	Qt5Widgetsd.dll!00007ffefb54c18d()	Unknown
 	Qt5Widgetsd.dll!00007ffefb2ec0d2()	Unknown
 	Qt5Widgetsd.dll!00007ffefb4a15fa()	Unknown
 	Qt5Widgetsd.dll!00007ffefb28c15e()	Unknown
 	Qt5Widgetsd.dll!00007ffefb288048()	Unknown
 	Qt5Cored.dll!00007ffef790e236()	Unknown
 	Qt5Cored.dll!00007ffef790e2dc()	Unknown
 	Qt5Widgetsd.dll!00007ffefb336f5d()	Unknown
 	Qt5Widgetsd.dll!00007ffefb335466()	Unknown
 	Qt5Widgetsd.dll!00007ffefb28c15e()	Unknown
 	Qt5Widgetsd.dll!00007ffefb286bff()	Unknown
 	Qt5Cored.dll!00007ffef790e236()	Unknown
 	Qt5Cored.dll!00007ffef790e0db()	Unknown
 	Qt5Guid.dll!00007ffef8311194()	Unknown
 	Qt5Guid.dll!00007ffef8313f13()	Unknown
 	Qt5Guid.dll!00007ffef82e2335()	Unknown
 	qwindowsd.dll!00007fff081205d2()	Unknown
 	Qt5Cored.dll!00007ffef79cf008()	Unknown
 	user32.dll!00007fff59c863ed()	Unknown
 	user32.dll!00007fff59c85de2()	Unknown
 	Qt5Cored.dll!00007ffef79cf8f8()	Unknown
 	qwindowsd.dll!00007fff08120584()	Unknown
 	Qt5Cored.dll!00007ffef7908463()	Unknown
 	Qt5Cored.dll!00007ffef790869e()	Unknown
 	Qt5Cored.dll!00007ffef790bd2f()	Unknown
 	Qt5Guid.dll!00007ffef830e518()	Unknown
 	Qt5Widgetsd.dll!00007ffefb2865aa()	Unknown
 	GPXLab.exe!main(int argc, char * * argv) Line 51	C++
 	GPXLab.exe!WinMain(HINSTANCE__ * __formal, HINSTANCE__ * __formal, char * __formal, int __formal) Line 97	C++

```

**Comment by BourgeoisLab** [comment]
Fixed with c5a07f1c3f687b0ce560fa593cdf4763c9876fe8.
